### PR TITLE
add dual stack support for AWS

### PR DESCRIPTION
**Pull request by PratikDeoghare**
**What this PR does / why we need it**:
Adds dual-stack support for AWS. 

**Which issue(s) this PR fixes** 

Implements part of https://github.com/kubermatic/kubermatic/issues/9100

**Optional Release Note**:

```release-note
add dual-stack support for AWS
```

### CODE REVIEW

File: pkg/cloudprovider/provider/aws/provider.go [hunk 1]
```diff
@@ -810,6 +823,10 @@ func (p *provider) Create(machine *clusterv1alpha1.Machine, data *cloudprovidert
 		},
 	}
 
+	if pc.Network.GetNetworkFamily() == util.DualStack {
```
Comment (rastislavs) [inline pkg/cloudprovider/provider/aws/provider.go:826]: what about `util.IPv6` here?

Comment (PratikDeoghare) [reply pkg/cloudprovider/provider/aws/provider.go:826]: My bad. Fixing.

**Review by moadqassem — APPROVED** [review]
/approve
/lgtm

**Pull request merged**